### PR TITLE
surface-control: init at 0.3.1-1

### PR DESCRIPTION
**Pull request by winterqt**
###### Motivation for this change

This adds [surface-control](https://github.com/linux-surface/surface-control), a command line tool to control various aspects of the Microsoft Surface line of devices from the command line. This is added with the intent of adding this package to the `nixos-hardware` configuration for these devices. (see https://github.com/NixOS/nixos-hardware/pull/227)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/surface-control/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ rustPlatform, fetchFromGitHub, lib }:
```
Comment (danieldk) [inline pkgs/applications/misc/surface-control/default.nix:1]: Style: put `lib` first.

**Review by danieldk — CHANGES_REQUESTED** [review]
Thank you for contributing to nixpkgs :+1: 

The program generates completion files during the build:

https://github.com/linux-surface/surface-control/blob/4ae4afab421e2bd8cac652072adba5203bc5d07a/build.rs#L13

It would be nice if they are installed too. You can use `installShellFiles` to install them, see the hooks section of the nixpkgs manual:

https://nixos.org/manual/nixpkgs/stable/#ssec-setup-hooks

**Comment by winterqt** [comment]
@danieldk Added shell completions, thanks for catching that.

Do you mind weighing in on https://github.com/NixOS/nixos-hardware/pull/227#discussion_r569026657 before this is merged? I assume the first option mentioned (getting the udev rule in the source tree of upstream) is probably the best option, but for a udev rule that will (probably) barely change, if ever?

**Comment by winterqt** [comment]
I managed to rebase successfully!

**Comment by danieldk** [comment]
Pushed some changes:

* Add some line breaks in `postInstall` to avoid overly long lines.
* Simplify `install` command and set permissions.

**Review by danieldk — APPROVED** [review]
LGTM, binary works.

Result of `nixpkgs-review pr 111690` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>surface-control</li>
  </ul>
</details>

**Pull request merged**